development/webpack/.eslintrc.js

```diff
@@ -0,0 +1,41 @@
+// this file is named .eslintrc.js because eslint checks for that file first
```

Comment:
I find so many of our eslint rules cripple JavaScript in absolutely absurd ways, so I've disabled some of the rules that I don't think make sense for the code in the webpack directory.

Reply:
hmm... I'm a little skeptical of this. I think eslint exception should happen when it prevents us from writing code that is the only way to achieve something. But some of these exceptions seem not strictly necessary.

You very well may be right that we should remove/change many of these rules, but then we should make a decision about removing / changing them for the whole code base. If we just make exceptions all over the place, then it makes a mess and does not fix the root cause of the problem.

Now, you will say "well we already have inline exceptions all over the place". Which is true. And is bad. But I think in most cases they happened because they seemed necessary for getting the code to do something that couldn't be done otherwise, whereas "sometimes its nice to do things like `something = else = null;`", "if (condition) return;` is useful for early returns without adding noise.", "`void` is useful to ignore return values" sound like preferences that are not strictly necessary.

Don't get me wrong, these preferences might be the best ones and we should default to them. But then let's agree on that collectively and apply to the whole codebase.

I know you find some eslint rules annoying, but if we think about the state we'd be in if there were 0 eslint rules, it would be very very bad. And so many code reviews would be full of people debating nits in code style.

With all of the above said, I don't really want to block this PR on this, so I don't think I will request changes here. But I think this compels me to prioritize making the conversation you want us to have about eslint rules happen. Then we can collectively decide which rules are in and which are out, and then we can also collectively work to eliminate exceptions, which may then require us to return to this code and refactor it a bit in the future.

Reply:
:heavy_check_mark: